### PR TITLE
Empty settings, jackett and performance improvements

### PR DESCRIPTION
**Pull request by Gaisberg**
- Implement possibility to start app with default settings
- Along with this came service_manager and observable utils
- Optimized cpu heavy parts of the code
- Spoked added jackett support

FRONTEND WILL NEED FIXING!

### CODE REVIEW

File: backend/utils/default_settings.json [hunk 1]
```diff
@@ -15,6 +17,10 @@
     "api_key": "",
     "update_interval": 80
   },
+  "jackett": {
+    "url": "",
```
Comment (dreulavelle) [inline backend/utils/default_settings.json:21]: thats weird.. this shouldnt be empty on the url.

Comment (Gaisberg) [reply backend/utils/default_settings.json:21]: Oops, yeah that needs to be corrected ;)

File: backend/utils/default_settings.json [hunk 2]
```diff
@@ -2,8 +2,10 @@
   "version": "0.3.0",
```
Comment (dreulavelle) [inline backend/utils/default_settings.json:2]: We need to be raising the version number on updates. Also, should have version number displayed in log during initialization

**Comment by AyushSehrawat** [comment]
Frontend has been fixed, need to fix some minor issue either in this PR or next one

```
^CException ignored in: <module 'threading' from '/home/mini/.pyenv/versions/3.12.0/lib/python3.12/threading.py'>
Traceback (most recent call last):
  File "/home/mini/.pyenv/versions/3.12.0/lib/python3.12/threading.py", line 1602, in _shutdown
    lock.acquire()
KeyboardInterrupt: 

```

Graceful shutdown of application

**Review by dreulavelle — APPROVED** [review]
Frontend works, and backend works. Merging and deleting branch!

**Pull request merged**